apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -100,6 +110,79 @@ export const pageRouter = router({
         ...siteMeta,
       }
     }),
+
+  reorderBlock: protectedProcedure
+    .input(reorderBlobSchema)
+    .mutation(async ({ input: { pageId, from, to, blocks, siteId }, ctx }) => {
+      // NOTE: we have to check against the page's content that we retrieve from db
+      // we adopt a strict check such that we allow the update iff the checksum is the same
+      const fullPage = await getFullPageById({ resourceId: pageId, siteId })
+
+      if (!fullPage?.content) {
+        // TODO: we should probably ping on call
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message:
+            "Unable to load content for the requested page, please contact Isomer Support",
+        })
+      }
+
+      const actualBlocks = fullPage.content.content
+
+      if (!isEqual(blocks, actualBlocks)) {
+        throw new TRPCError({
+          code: "CONFLICT",
+          message:
+            "Someone on your team has changed this page, refresh the page and try again",
+        })
+      }
+
+      if (
+        from >= actualBlocks.length ||
+        to >= actualBlocks.length ||
+        from < 0 ||
+        to < 0
+      ) {
+        // NOTE: If this happens, this indicates that either our dnd libary on our frontend has a
+        throw new TRPCError({ code: "UNPROCESSABLE_CONTENT" })
+      }
+
+      const [movedBlock] = actualBlocks.splice(from, 1)
+      if (!movedBlock) return blocks
+      if (!fullPage.draftBlobId || !fullPage.mainBlobId) {
+        throw new TRPCError({ code: "NOT_FOUND" })
```

Comment:
would a msg be useful here?

Reply:
Also generic comment, we explored structuring comments in one of our early RFCs on Isomer. For future implementation, but we can think about having internal error codes which we can map to for errors

Reply:
sure, added a message.